protocol/Makefile

```diff
@@ -1,47 +1,48 @@
 default: build
 all: build
 
-ICED=node_modules/.bin/iced
 AVDL2JSON=node_modules/.bin/avdl2json
 AVDLC=node_modules/.bin/avdlc
 
-json/%.json: avdl/%.avdl
+json/%.json: $(AVDL2JSON) avdl/%.avdl
 	$(AVDL2JSON) -i $< -o $@~ && mv $@~ $@
 
-config:
+$(AVDL2JSON):
+	npm i
+$(AVDLC):
 	npm i
 
 # By default, all avdl/*.avdl are considered and made into outputs
 # as long as they have messages in them. BUT you can override this
 # check via -f. So the two files specified here via -f don't have
 # messages but should be made into .json's...
-build-stamp: avdl/*.avdl $(AVDL2JSON) | config
+build-stamp: avdl/*.avdl | $(AVDL2JSON)
```

Comment:
Changing to a non-phony dependency means that `npm i` won't be called if `node_modules/.bin/avdl2json` (or `node_modules/.bin/avdlc`) already exists.